src/Modules/ModuleTypes.jl

```diff
@@ -497,12 +464,100 @@ The type of decorated free module elements. An element of a decorated free modul
 given by a sparse row (`SRow`) which specifies its coordinates with respect to the basis
 of standard unit vectors of $F$.
 """
-struct FreeModElem_dec{T} <: AbstractFreeModElem_dec{T}
+struct FreeModElem_dec{T} <: AbstractFreeModElem{T}
   coords::SRow{T} # also usable via coeffs()
   parent::FreeMod_dec{T}
 
   function FreeModElem_dec{T}(coords::SRow{T}, parent::FreeMod_dec{T}) where T
     r = new{T}(coords,parent)
     return r
   end
+end
+
+
+
+
+
+#abstract type ModuleFP_dec{T} <: ModuleFP{T} end
+
+#abstract type ModuleFPElem_dec{T} <: ModuleFPElem{T} end
+const ModuleFP_dec{T} = Union{FreeMod_dec{T}} # SubQuo_dec{T} will be included
+const ModuleFPElem_dec{T} = Union{FreeModElem_dec{T}} # SubQuoElem_dec{T} will be included
+
+#const AbstractFreeMod{T} = Union{FreeMod{T}, FreeMod_dec{T}}
+#const AbstractFreeModElem{T} = Union{FreeModElem{T}, FreeModElem_dec{T}}
+
+#const AbstractSubQuo{T} = Union{SubQuo{T}}
+#const AbstractSubQuoElem{T} = Union{SubQuoElem{T}}
```

Comment:
What about this? Can it be removed or is the plan to un-comment this in the future?